src/p2p/Sync.ts

```diff
@@ -551,154 +552,171 @@ export function totalNodeCount(cycle: P2P.CycleCreatorTypes.CycleRecord) {
-      apoptosized: 'a',
-      networkDataHash: 'a',
-      networkReceiptHash: 'a',
-      networkSummaryHash: 'a',
-      desired: 'n',
-    })
-    if (err) {
-      warn('Type validation failed for cycleRecord: ' + err)
-    }
-    for (const refreshedArchiver of cycleRecord.refreshedArchivers) {
-      err = validateTypes(refreshedArchiver, archiverType)
-      if (err) {
-        warn('Validation failed for cycleRecord.refreshedArchivers: ' + err)
-        return false
-      }
-    }
-    for (const refreshedConsensor of cycleRecord.refreshedConsensors) {
-      err = validateTypes(refreshedConsensor, {
-        curvePublicKey: 's',
-        status: 's',
-      })
-      if (err) {
-        warn('Validation failed for cycleRecord.refreshedConsensors: ' + err)
-        return false
-      }
-    }
-    for (const joinedArchiver of cycleRecord.joinedArchivers) {
-      err = validateTypes(joinedArchiver, archiverType)
-      if (err) {
-        warn('Validation failed for cycleRecord.joinedArchivers: ' + err)
-        return false
-      }
-    }
-    for (const leavingArchiver of cycleRecord.leavingArchivers) {
-      err = validateTypes(leavingArchiver, archiverType)
-      if (err) {
-        warn('Validation failed for cycleRecord.leavingArchivers: ' + err)
-        return false
-      }
-    }
-    for (const joinedConsensor of cycleRecord.joinedConsensors) {
-      err = validateTypes(joinedConsensor, {
-        cycleJoined: 's',
-        counterRefreshed: 'n',
-        id: 's',
-      })
-      if (err) {
-        warn('Validation failed for cycleRecord.joinedConsensors: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.activated) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.refreshedConsensors: ' + err)
-        return false
-      }
-    }
-    for (const activatedPublicKey of cycleRecord.activatedPublicKeys) {
-      if (typeof activatedPublicKey !== 'string') {
-        warn('Validation failed for cycleRecord.activatedPublicKeys: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.lost) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.lost: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.refuted) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.refuted: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.joined) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.joined: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.returned) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.returned: ' + err)
-        return false
-      }
+  if (config.p2p.useAjvCycleRecordValidation) {
+    const errors = verifyPayload(AJVSchemaEnum.CycleRecords, cycles)
```

Comment:
I would argue that returning partial valid data here could cause to more issues later down the line. So if we validate a bunch of cycle records 1 by 1 maybe one of them is failing but that could lead to invalid node lists and now with network generated txs that could lead to missed txs.

Reply:
I'll add a try/catch to this. What do you think of doing the try/catch in `verifyPayload` function itself? we already return an Error if something through validation. We could handle that the same way

Reply:
consider a scenario.. where i have a single malicious node.. that always returns.. bad or incorrect format in getcyles ( query .. i.e. sync-cycles ).. so this will result in ajv validation being failed every time.. just because of 1 malicious active node present in the network.. and the sync-cycles step would never succeed. @PudgyPug

Reply:
> and the part of force casting can run into runtime-exceptions
could you elaborate on this? Casting in typescript is not causing runtime exceptions

Reply:
> consider a scenario.. where i have a single malicious node.. that always returns.. bad or incorrect format in getcyles ( query .. i.e. sync-cycles ).. so this will result in ajv validation being failed every time.. just because of 1 malicious active node present in the network.. and the sync-cycles step would never succeed. @PudgyPug

robustQuery is taking care of that

Reply:
cleared up over huddle. try/catch has been added to validation

Reply:
based on the conversations in dm.. and also as per my observations ( we are using robsutQuery to query the nodes )  : Redundancy is the minimum number of consistent responses needed, not the majority of the entire network. robustQuery doesn’t query all nodes in the network at once. Instead, it queries a subset of nodes and looks for redundancy within that subset and case where [ validCycleRecord, notValidCycleRecord ] will be present will be less likely because of the no. of good nodes present in the network will be more.. so adding try/catch should suffice in this case beside the ajv validation with response ( in the parent function getCycles )..